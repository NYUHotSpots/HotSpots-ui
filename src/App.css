.root {
  height: 100%;
  width: 100%;
  padding: 0;
  margin: 0;
  display: flex;
  justify-content: center;
  align-items: center;

}

.quote {
  font-style: italic;
}

.intro {
  text-align: left;
}

.content {
  width: 800px;
  display: flex;
  flex-direction: column;
  /* align-items: center; */
  gap: 2rem;
}

.page-button {
  min-width: 100px;
  max-width: 200px;
  border: 1px solid #94A3B8;
  background-color: #E2E8F0;
  padding-right: 25px;
  padding-left: 25px;
  padding-top: 10px;
  padding-bottom: 10px;
  border-radius: 5px;
  cursor: pointer;
  margin-left: 5px;
}

.button {
  padding-right: 15px;
  padding-top: 5px;
  padding-bottom: 5px;
  padding-left: 15px;
  border: 1px solid #94A3B8;
  background-color: #E2E8F0;
  border-radius: 5px;
  cursor: pointer;
}

.page-button:hover {
  background-color: #CBD5E1;
}

.button-container {
  display: flex;
  justify-content: flex-end;
}

.grid-container {
  display: grid;
  grid-template-columns: repeat(3, 1fr);
  gap: 10px;
}

@media (max-width: 600px){
  .grid-container{
    grid-template-columns: repeat(1, 1fr);
    gap: 10px;
  }
}