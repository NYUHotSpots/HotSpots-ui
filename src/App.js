import React from 'react';
import {HashRouter as Router, Route, Switch} from 'react-router-dom';

import Home from './pages/Home/Home';
import SpotInfo from './pages/SpotInfo/SpotInfo';
import UpdateSpotInfo from './pages/UpdateSpotInfo/UpdateSpotInfo';

import './App.css';
import CheckReviews from './pages/CheckReviews/CheckReviews';
import CreateReview from './pages/CreateReview/CreateReview';
import ViewReviews from './pages/ViewReviews/ViewReviews';
import UpdateSpotFactors from './pages/UpdateSpotFactors/UpdateSpotFactors';

import CreateSpot from './pages/CreateSpot/CreateSpot';

import SubmissionSuccess from './pages/SubmissionSuccess/SubmissionSuccess';

import ProtectedRoute from './components/ProtectedRoute/ProtectedRoute';
import Profile from './pages/Authentication/Profile';

function App() {
  return (
    <div className="root">
      <div className="content">
        <Router basename={process.env.PUBLIC_URL}>
          <Switch>
            <Route exact={true} path={'/'}>
              <Home />
            </Route>

            <Route exact={true} path={'/spots/:spotID'}>
              <SpotInfo />
            </Route>


            <Route exact={true} path={'/CheckReviews'}>
              <CheckReviews />
            </Route>

            <Route exact={true} path={'/spot_review/read/:spotID'}>
              <ViewReviews />
            </Route>

            <Route exact={true} path={'/createreview/:spotID'}>
              <CreateReview />
            </Route>

            {/* <Route exact={true} path={'/createspot'}>
              <CreateSpot />
            </Route> */}
            <ProtectedRoute path="/createspot" component={CreateSpot} />

            <Route exact={true} path={'/updatespotinfo'}>
              <UpdateSpotInfo />
            </Route>

            <Route exact={true} path={'/updatespotfactors/:spotID'}>
              <UpdateSpotFactors />
            </Route>

            <Route exact={true} path={'/submissionsuccess'}>
              <SubmissionSuccess />
            </Route>

            <ProtectedRoute path="/profile" component={Profile} />
          </Switch>
        </Router>
      </div>
    </div>
  );
}

export default App;
