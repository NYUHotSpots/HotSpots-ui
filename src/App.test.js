import { render, screen } from "@testing-library/react";
import renderer from "react-test-renderer";
import App from "./App";

describe("App Component", () => {
  test("renders learn react link", () => {
    render(<App />);
    const linkElement = screen.getByText(/HotSpots/i);
    expect(linkElement).toBeInTheDocument();
  });

  test("snapshots", () => {
    const tree = renderer.create(<App />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
