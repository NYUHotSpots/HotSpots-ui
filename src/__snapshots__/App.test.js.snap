// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Component App snapshots 1`] = `
<div
  className="root"
>
  <div
    className="content"
  >
    <body>
      <div
        className="home_header"
      >
             
        <h1>
          HotSpots
        </h1>
        <button
          className="page-button"
          onClick={[Function]}
        >
          Log In / Sign Up
        </button>
      </div>
      <div
        className="intro"
      >
        <p
          className="quote"
        >
          "Crowdsourcing study spaces for students by students"
        </p>
        <p
          className="dev-names"
        >
          Tiffany Chan, Kevin Chen, Kevin Iza, Kathy Pan
        </p>
      </div>
      <div
        className="button-container"
      >
        <button
          className="page-button"
          onClick={[Function]}
        >
          Add New Spot
        </button>
      </div>
      <div
        className="grid-container"
      >
        <div
          className="rooms-empty"
        >
          <p>
            Sorry there are no Locations right now... Come back later 
          </p>
        </div>
      </div>
    </body>
  </div>
</div>
`;
